searchlib/src/tests/common/bitvector/bitvector_test.cpp

```diff
@@ -907,4 +1008,77 @@ TEST(BitvectorTest, dynamic_guard_bits)
     g.reclaim(2);
 }
 
+TEST_F(BitVectorTest, read_from_attriute_vector_file)
+{
+    vespalib::GenerationHolder g;
+    std::string bvpath(testdata + "/bv");
+    FastOS_File file;
+    file.OpenReadOnly(bvpath.c_str());
+    ASSERT_TRUE(file.IsOpened());
+    GrowableBitVector bv(1, 1, g);
+    bv.writer().clear();
+    uint32_t numDocs = _file_bv->size();
+    bv.extend(numDocs);
+    auto entry_size = BitVector::legacy_num_bytes_with_single_guard_bit(numDocs);
+    ASSERT_LE(entry_size, bv.writer().numBytes(bv.writer().size()));
+    file.ReadBuf(bv.writer().getStart(), entry_size);
+    bv.fixup_after_load();
+    auto bv_snap = bv.make_snapshot(bv.writer().size());
+    EXPECT_TRUE(*bv_snap == *_file_bv);
+    g.assign_generation(1);
+    g.reclaim(2);
+}
+
+class BitVectorCompatibilityTest : public BitVectorTest,
+                                   public testing::WithParamInterface<uint32_t> {
+protected:
+    BitVectorCompatibilityTest();
+    ~BitVectorCompatibilityTest() override;
+};
+
+BitVectorCompatibilityTest::BitVectorCompatibilityTest()
+    : BitVectorTest(),
+      testing::WithParamInterface<uint32_t>()
+{
+}
+
+BitVectorCompatibilityTest::~BitVectorCompatibilityTest() = default;
+
+INSTANTIATE_TEST_SUITE_P(BitVectorCompatibilityTest, BitVectorCompatibilityTest, testing::Values(1, 2),
+                         param_as_string);
+
+TEST_P(BitVectorCompatibilityTest, read_from_file_bitvector_dictionary_file)
```

Comment:
Consider renaming test case to mention interaction with guard bits (alternatively, rename the test class instead (`BitVectorGuardBitCompatibilityTest`?))